rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/parsing/facade/compilerDirectives.kt

```diff
@@ -71,96 +74,153 @@ fun String.resolveCompilerDirectives(): String {
     val definitions = mutableListOf<String>()
     var useRow = true
     var eofInvoked = false
+    var directiveRow = false
     // Split input text into rows
     val rows = this.lines()
 
     // Check each row against regex patterns
     for ((index, row) in rows.withIndex()) {
-        when {
-            IF_DEFINED_PATTERN.matches(row) -> {
-                // Control if IF_DEFINED is acceptable
-                val acceptedLastCodes = listOf(SYN_RELEVANT_DIRECTIVES.NONE, SYN_RELEVANT_DIRECTIVES.ENDIF)
-                if (!acceptedLastCodes.any { it == lastCode }) {
+        // if eof invoked manage all remaining rows as comments
+        if (!eofInvoked) {
+            when {
+                IF_DEFINED_PATTERN.matches(row) -> {
+                    directiveRow = true
+                    // Control if IF_DEFINED is acceptable
+                    val acceptedLastCodes = listOf(SYN_RELEVANT_DIRECTIVES.NONE, SYN_RELEVANT_DIRECTIVES.ENDIF)
+                    if (!acceptedLastCodes.any { it == lastCode }) {
                         val exc = CompilerDirectivesException("Unexpected IF_DEFINED directive at line " + (index + 1))
                         throw AstCreatingException(this, exc).fireErrorEvent(null)
```

Comment:
Throw a `CompilerDirectivesException` instead of firing of `AstCreatingException`. `fireErrorEvent` meant to make some parsing and ast creatiing errors not blocking, but if we have an error in this stage it must be blocking

Reply:
Done